test/benchmarks/mongoBench/suites/parallelBench.js

```diff
@@ -0,0 +1,206 @@
+const { createReadStream, createWriteStream } = require('fs');
+const { rm, mkdir, readdir } = require('fs/promises');
+const { resolve } = require('path');
+const { Readable, once } = require('stream');
+const readline = require('readline');
+const {
+  makeClient,
+  disconnectClient,
+  dropDb,
+  initBucket,
+  dropBucket,
+  initCollection,
+  initDb,
+  connectClient,
+  createCollection
+} = require('../../driverBench/common');
+const { pipeline } = require('stream/promises');
+const { EJSON } = require('bson');
+
+const benchmarkFileDirectory = resolve(__dirname, '..', '..', 'driverBench', 'spec', 'parallel');
+
+async function initTemporaryDirectory() {
+  const temporaryDirectory = resolve(benchmarkFileDirectory, 'downloads');
+  await rm(temporaryDirectory, { recursive: true, force: true }).catch(() => {
+    // ignore errors, specifically directory not found
+  });
+  await mkdir(temporaryDirectory);
+  this.temporaryDirectory = temporaryDirectory;
+}
+
+async function clearTemporaryDirectory() {
+  const fileNames = await readdir(this.temporaryDirectory);
+  const files = fileNames.map(filename => resolve(this.temporaryDirectory, filename));
+
+  await Promise.all(files.map(file => rm(file)));
+}
+
+async function removeTemporaryDirectory() {
+  await rm(this.temporaryDirectory, { recursive: true, force: true }).catch(() => {
+    // ignore errors, specifically directory not found
+  });
+}
+
+async function ldjsonMultiUpload() {
+  const directory = resolve(benchmarkFileDirectory, 'ldjson_multi');
+  const files = await readdir(directory);
+  const uploads = files.map(async file => {
+    const stream = createReadStream(resolve(directory, file));
+    const lineReader = readline.createInterface({
+      input: stream
+    });
+
+    const bulkWrite = this.collection.initializeUnorderedBulkOp();
+
+    for await (const line of lineReader) {
+      bulkWrite.insert(JSON.parse(line));
+    }
+
+    stream.close();
+    lineReader.close();
+
+    return bulkWrite.execute();
+  });
+
+  await Promise.all(uploads);
+}
+
+async function ldjsonMultiExport() {
+  const skips = Array.from({ length: 100 }, (_, index) => index * 5000);
+  const collection = this.collection;
+  const temporaryDirectory = this.temporaryDirectory;
+
+  const promises = skips.map(async function (skip) {
+    const documentCursor = collection.find({}, { skip, limit: 5000 });
+    documentCursor.map(doc => EJSON.stringify(doc));
```

Comment:
If we're going to test using streams using streams it seems like it would be more relavent to do the transform as yet another stream:
```ts
pipeline(input, ejsonify, output)
```
Thoughts?

Reply:
internally that's what happens anyways when you use cursor.map + stream, so I don't think it's necessary.  If you want I can explicitly construct a transform stream though

Reply:
nope, good call let's stick with what's built in